src/main/java/com/amazon/aws/iot/greengrass/component/common/ComponentConfiguration.java

```diff
@@ -22,6 +24,14 @@
 
     JsonNode defaultConfiguration;
 
+    JsonNode defaultParameters;
+
+    Map<String, TemplateParameter> parameterSchema;
+
+    public ComponentConfiguration(JsonNode defaultConfiguration) {
+        this.defaultConfiguration = defaultConfiguration;
+    }
```

Comment:
why is this added?

Reply:
the thought is to not break any existing applications that use the `AllArgsConstructor` to create an object

Reply:
Does anything use allargs?

Reply:
I have no idea. But at the same time, why would we have `NoArgs` and `AllArgs` if we never expect anyone to use them?

Reply:
They exist for Jackson. Jackson needs them both.

Reply:
I see. But this doesn't address the question of constructor usage. You probably know better than me: is it 100% guaranteed that between the time this PR is merged into my fork and the time it makes it into the main repo, nobody will try to use construct a ComponentConfiguration without using the builder?

Reply:
I suspect the answer is no, which is why I'm hesitant

Reply:
No, but we can fix it as needed. We always use builders, so it is highly highly unlikely that this will be an issue. No one constructs these objects manually anyway, it is always Jackson.

Reply:
Alright. Good to know